assets/src/stories-editor/blocks/amp-story-page/copy-paste-handler.js

```diff
@@ -123,24 +125,36 @@ export default withDispatch( ( dispatch, ownProps, { select } ) => {
 			clearCopiedMarkup();
 			return;
 		}
-		const serialized = serialize( getBlocksByClientId( selectedBlockClientIds ) );
-		setCopiedMarkup( serialized );
-
-		if ( isCut ) {
-			// TODO: Remove selected Blocks.
-			// The code below works most of the time, but sometimes (unable to tell when and why) another element on page is selected when the block is removed, and then the browser copies this other element in stead of the previously selected element (that has now been removed).
-			/* for ( const clientId of selectedBlockClientIds ) {
-				removeBlock( clientId );
-			} */
-			// wrapping the call in setTimeout fixes the case where another element is selected on cut, but throws an error in the cases, where the above code works fine.
+
+		// Don't allow story blocks to be copyied.
+		for ( const selectedBlockClientId of selectedBlockClientIds ) {
+			if ( isStoryBlock( selectedBlockClientId ) ) {
+				clearCopiedMarkup();
+				return;
+			}
 		}
-	};
 
-	const onCopy = createCutCopyHandler( false );
-	const onCut = createCutCopyHandler( true );
+		const copyBlocks = getBlocksByClientId( selectedBlockClientIds );
+		const serialized = serialize( copyBlocks );
+		// Workout what type of event, from event object passed to this function.
+		const isCut = ( event.type === 'cut' );
+
+		// Make sure that setCopiedMarkup finishes before doing anything else.
+		setCopiedMarkup( serialized ).then( () => {
+			copyTextToClipBoard( serialized );
+
+			if ( isCut ) {
+				const pageClientId = getCurrentPage();
+				for ( const clientId of selectedBlockClientIds ) {
+					// On removing block, change focus to the page, to make sure that editor doesn't get confused.
+					selectBlock( pageClientId );
```

Comment:
We have to do this inside the loop? Or is the loop actually a little superfluous, as we can only ever have a single block selected?

Reply:
Doing this in the loop, makes sure that if multiple blocks are copied, that the editor doesnt get confused and try and select a block that has been removed. Putting the code here, is not an issue, as most of the time, only one block is being copied anyway..

Reply:
> "[...] try and select a block that has been removed"

Sounds like something that should be added as an inline doc.

Just "to make sure that editor doesn't get confused" alone is not very helpful.

Reply:
:100: